src/include/duckdb/storage/compression/alprd/alprd_analyze.hpp

```diff
@@ -0,0 +1,147 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/storage/compression/alprd/alprd_analyze.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "duckdb/function/compression_function.hpp"
+#include "duckdb/storage/compression/patas/patas.hpp"
+#include "duckdb/storage/compression/alprd/alprd.hpp"
+#include "duckdb/storage/compression/alprd/alprd_constants.hpp"
+#include "duckdb/storage/compression/alp/alp_constants.hpp"
+
+#include <cmath>
+
+namespace duckdb {
+
+template <class T>
+struct AlpRDAnalyzeState : public AnalyzeState {
+public:
+	using EXACT_TYPE = typename FloatingToExact<T>::type;
+
+	AlpRDAnalyzeState() : state((void *)this) {
+	}
+
+	idx_t vectors_count = 0;
+	idx_t total_values_count = 0;
+	idx_t vectors_sampled_count = 0;
+	vector<EXACT_TYPE> rowgroup_sample;
+	AlpRDState<T, true> state;
+};
+
+template <class T>
+unique_ptr<AnalyzeState> AlpRDInitAnalyze(ColumnData &col_data, PhysicalType type) {
+	return make_uniq<AlpRDAnalyzeState<T>>();
+}
+
+/*
+ * ALPRD Analyze step only pushes the needed samples to estimate the compression size in the finalize step
+ */
+template <class T>
+bool AlpRDAnalyze(AnalyzeState &state, Vector &input, idx_t count) {
```

Comment:
Yes, initially we wanted to only do the analyze once. However (as we explain in the PR description), we decided to make ALP and ALPRD two different algorithms inside duckdb to improve code readability. Furthermore, they are two very different approaches, and we think that in the context of duckdb, we should be able to force compression using either of them.

Using exactly the same code for the Analyse will be a bit tricky since i) one of them load values as EXACT_TYPE, and the other as T; and ii) One of them stores also the complete vector from which we take samples while the other only store the samples. 

However we have tried to merge the common behaviours in a new `alp::AlpUtils` class (i.e. checking wether we have to take samples from the current vector, obtaining the sampling parameters and replacing the null values).